Kudu.Services.Web/App_Start/NinjectServices.cs

```diff
@@ -628,6 +629,23 @@ private static void RemoveTempFileFromUserDrive(IEnvironment environment)
             FileSystemHelpers.DeleteDirectorySafe(Path.Combine(environment.RootPath, "data", "Temp"), ignoreErrors: true);
         }
 
+        // add/update gitconfig file to be equivalent of running
+        // git config --global --add safe.directory "*"
+        private static void EnsureGitConfigFileWithSafeDirectoryConfig(IEnvironment environment)
+        {
+            var markerFilePath = Path.Combine(environment.RootPath, "gitsafedirectory.marker");
+            if (File.Exists(markerFilePath))
+            {
+                return;
+            }
+            OperationManager.SafeExecute(() =>
+            {
+                var gitConfigPath = Path.Combine(environment.RootPath, ".gitconfig");
+                File.AppendAllLines(gitConfigPath, new[] {"[safe]", "    directory = *" });
```

Comment:
what if customer already have the file with [safe] defined?

Reply:
since we are appending this at the end of the file, `[safe] directory=*` takes precedence even if some other `[safe]` value is present in the gitconfig file

Reply:
Do you see any potential issues with marking all directories as safe? The user has ownership of all the directories and if they have their own `[safe]` configuration added in the gitconfig, their directory list would be a subset of `*` directories, so it should not impact their usage.

Reply:
I think it is fine as long as we tested it (duplicate [safe] sections in gitconfig).   After all it will be behind off-by-default knob.